packages/discovery-provider/src/tasks/entity_manager/entities/track.py

```diff
@@ -620,6 +625,38 @@ def delete_track(params: ManageEntityParameters):
     params.add_record(track_id, deleted_track)
 
 
+def download_track(params: ManageEntityParameters):
+    validate_track_tx(params)
```

Comment:
do we need to validate that the params.user_id is the signer too?

Reply:
Good q. `validate_track_tx` calls `validate_signer`, so this does. Do we need that? Arguably no. I designed everything else assuming that there _isn't_ a user here, so if I wanted to be consistent I would need to make sure that a user not being found is allowed.

HOWEVER, in the client, we don't allow downloads unless signed in. So I think my assumption was off - despite the download API not requiring a user (iirc), the client does. Maybe the right thing to do is require an account for the download API and then keeping everything requiring a user?

Reply:
chatted offline. kinda whatever you want to do here. require user is fine